presto-main/src/main/java/com/facebook/presto/metadata/StaticFunctionNamespaceStore.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.metadata;
+
+import com.google.common.base.Splitter;
+import com.google.common.collect.ImmutableList;
+import io.airlift.log.Logger;
+
+import javax.inject.Inject;
+
+import java.io.File;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import static com.facebook.presto.util.PropertiesUtil.loadProperties;
+import static com.google.common.base.Preconditions.checkState;
+
+public class StaticFunctionNamespaceStore
+{
+    private static final Logger log = Logger.get(StaticFunctionNamespaceStore.class);
+    private static final Splitter SPLITTER = Splitter.on(',').trimResults().omitEmptyStrings();
+
+    private final FunctionManager functionManager;
+    private final File configDir;
```

Comment:
Do we really need a directory? We have been gradually ditching file-based configs. The only place I remember we still use is the catalog directory that is used for installing connector plugins and the other file-based configs are legacy logic. If the config is just some attributes, shall we use the normal configuration than a file?

Reply:
I'm basically copying the configuration of catalogs here because the concepts kinda overlap so I think it's nice to keep them the same way. Happy to take other alternatives if they makes more sense. What do you mean by "normal configuration"?

Reply:
Maybe I should understand more about how function namespaces will be configured. Will it be something similar to connectors, where each connector has its own config? Or it will be something simple like `log.properties` below. The former may need a file-based config but the latter can be just put into a new class like `LogConfig.java`.

<img width="258" alt="Screen Shot 2019-08-30 at 1 54 06 PM" src="https://user-images.githubusercontent.com/2192913/64051629-8f074d80-cb30-11e9-84aa-5bf5dda6ce6c.png">

Reply:
The way I see this, each config would be configuring a different function namespace manager, which can manage 1 or many function namespaces. Some function namespaces are just logical separations but physically could be managed the same way. For example, you might want to have separate function namespaces for different teams, but they are all SQL functions and can be managed using the same manager with the same configuration. On the other hand, other function namespaces can be physically different. For example, you may want to support MySQL functions as a different function namespace, so people can directly use `mysql.function.foo` in their queries. This kind of function namespaces would have a 1:1 mapping to the connector they serve.

Let me know whether you think this is reasonable. We can discuss this further.